pkg/ccl/importccl/read_import_pgdump.go

```diff
@@ -578,11 +584,27 @@ func (m *pgDumpReader) readFile(
 				if count <= resumePos {
 					continue
 				}
-				if expected, got := len(conv.VisibleCols), len(tuple); expected != got {
+				if expected, got := expectedColLen, len(tuple); expected != got {
 					return errors.Errorf("expected %d values, got %d: %v", expected, got, tuple)
 				}
-				for i, expr := range tuple {
-					typed, err := expr.TypeCheck(ctx, &semaCtx, conv.VisibleColTypes[i])
+				conv.IsTargetCol = make(map[int]struct{})
+				colLookup := make(map[string]int, 0)
+				for j, col := range conv.VisibleCols {
+					colLookup[col.Name] = j
```

Comment:
This should be created (at most) once per insert, not once per insert tuple. Once per file would be better, but that may be tricky.

Reply:
I moved this to an outer loop so that it will be created per insert. Don't think it would work for once per file because in the case of `IMPORT PGDUMP blah...` the targeted columns could be different for each insert statement within the file.